### PR TITLE
Guard for changes in dictionary size during iteration.

### PR DESCRIPTION
**Pull request by cthulahoops**

### CODE REVIEW

File: vm/src/obj/objdict.rs [hunk 1]
```diff
@@ -313,25 +311,28 @@ macro_rules! dict_iterator {
         #[derive(Debug)]
         struct $iter_name {
             pub dict: PyDictRef,
-            pub position: Cell<usize>,
+            pub position: RefCell<dictdatatype::DictPosition>,
```
Comment (coolreader18) [inline vm/src/obj/objdict.rs]: If you made `DictPosition` `Copy`, you could keep this as a `Cell`.

Comment (coolreader18) [reply vm/src/obj/objdict.rs]: Actually, does this even need to be mutable? It looks like you're not changing it in `__next__`.

Comment (cthulahoops) [reply vm/src/obj/objdict.rs]: Thinking more on this, given the eventual implementation I use position and size in different places so there's no need to bundle them together.

File: vm/src/obj/objdict.rs [hunk 2]
```diff
@@ -315,11 +318,8 @@ impl Iterator for DictIter {
     type Item = (PyObjectRef, PyObjectRef);
 
     fn next(&mut self) -> Option<Self::Item> {
-        match self.dict.entries.borrow().next_entry(self.position) {
-            Some((new_position, key, value)) => {
-                self.position = new_position;
-                Some((key.clone(), value.clone()))
-            }
```
Comment (coolreader18) [inline vm/src/obj/objdict.rs]: You could just do `.cloned()` here.

Comment (cthulahoops) [reply vm/src/obj/objdict.rs]: I don't think it works through the tuple.

**Pull request merged**